### PR TITLE
Remove Chipyard submodule from FireSim (Chipyard designs must clone Chipyard 1st)

### PR DESCRIPTION
**Pull request by abejgonzalez**
This PR restructures FireSim to (mainly) be a library submodule of a larger project. As a consequence, it also updates all the documentation.

- If users want to use FireSim, they should clone Chipyard 1st and follow along with that documentation 1st.
- CI for building Chipyard target-specific bitstreams is moved to Chipyard
  - This means that the FireSim repo doesn't test anything related to Chipyard (bitstream builds, software metasim testing, etc). This repo only tests what's under it (midas, firesim-lib tests).
  - This also includes moving HWDBs and recipes to Chipyard.
  - All submodules related to Chipyard are removed
- Documentation changes
  - FireSim docs now properly point to Chipyard's HWDB and build recipes files
  - FireSim docs are re-organized a bit more for clarity - https://docs.fires.im/en/isolate-chipyard-support/index.html
  - FireSim docs (in `conf.py`) point to a specific version of Chipyard (to resolve links) but recommend checking out `main`
  - Adds initial docs on how to add a new FPGA and how to use FireSim as a library in a non-Chipyard top 
  - Documentation is now formatted with `docstrfmt` and checked in CI
- Build setup now adds a `FS_DIR` variable to help users navigate to the current FireSim top-level repo
- Relative path for the makefrag added in #1806 is fixed

Future Work
- Create a simplified non-Chipyard top-level example and add documentation for it
- Add more CI to test building bitstreams and running beefier simulations (like the Chipyard target simulations) from this repo.

Sister PR
ucb-bar/chipyard#2014

#### Contributor Checklist
- [ ] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [ ] Did you add Scaladoc/docstring/doxygen to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous prints/debugging code?
- [ ] Did you state the UI / API impact?
- [ ] Did you specify the Verilog / AGFI compatibility impact?

- [ ] If applicable, did you regenerate and publicly share default AGFIs?

- [ ] If applicable, did you apply the `Please Backport` label?

#### Reviewer Checklist (only modified by reviewer)
Note: to run CI on PRs from forks, comment `@Mergifyio copy main` and manage the change from the new PR.
- [ ] Is the title suitable for inclusion in the changelog and does the PR have a `changelog:<topic>` label?
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

#### CI Help
Add the following labels to modify the CI for a set of features.
Generally, a label added only affect subsequent changes to the PR (i.e. new commits, force pushing, closing/reopening).
See `ci:*` for full list of labels:
- `ci:fpga-deploy` - Run FPGA-based E2E testing
- `ci:local-fpga-buildbitstream-deploy` - Build local FPGA bitstreams for platforms that are released
- `ci:persist-prior-workflows` - Prevent prior CI workflows from automatically cancelling with subsequent changes
- `ci:disable` - Disable CI
- `ci:disable-scala-tests` - Disable Scala tests in CI

### CODE REVIEW

**Comment by joonho3020** [comment]
I have a dumb question. Lets say you add some change in FireSim and want to run the entire CI flow (bitstream builds + linux boot).

Would the process for that look like this:
- Make whatever change in FS
- Bump FS in CY & add `build-bitstream` & `linux-poweroff` tags in the CY PR?

In this case, how would we block the FireSim PR from getting in before the CI in CY passes?

**Comment by abejgonzalez** [comment]
> I have a dumb question. Lets say you add some change in FireSim and want to run the entire CI flow (bitstream builds + linux boot).
> 
> Would the process for that look like this:
> 
> * Make whatever change in FS
> * Bump FS in CY & add `build-bitstream` & `linux-poweroff` tags in the CY PR?
> 
> In this case, how would we block the FireSim PR from getting in before the CI in CY passes?

This is a very valid question. Your flow would be what you would do. There would be no way to block the PR in FireSim on what is happening in Chipyard. 

In an ideal world, the FireSim CI would check that the feature added would be tested separately from a Chipyard-based target simulation. For example, with a `midas` Scala test or something smaller than a Chipyard-SoC simulation.

For now, this PR removes the extra CI in favor of re-building it up again with FPGA testing of midasexamples, firesim-lib testing, etc with the manager. But I'll leave that to future PRs.

**Review by joonho3020 — APPROVED** [review]
LGTM

**Pull request merged**
